[iOS] Add size update for Shell Popup Overlay after rotation

cat0363 opened the pull request:
### Description of Change

<!-- Enter description of the fix in this section -->
When the terminal is rotated while the Flyout of Shell is displayed, the size of the overlay does not 
change from the size before rotation. In this PR, we will be updating the size of the overlay after 
rotation so that the overlay appears as intended.

Override the ViewWillTransitionToSize method to update the Tapoffview's Frame on rotation detection.
This will allow the size of the overlay portion to update on rotation. I needed to override the ViewWillTransitionToSize 
method, so I added the definition to PublicAPI.Unshipped.txt.

[src\Controls\src\Core\PublicAPI\net-ios\PublicAPI.Unshipped.txt]

    ~override Microsoft.Maui.Controls.Platform.Compatibility.ShellFlyoutRenderer.ViewWillTransitionToSize(CoreGraphics.CGSize toSize, UIKit.IUIViewControllerTransitionCoordinator coordinator) -> void

[src\Controls\src\Core\PublicAPI\net-maccatalyst\PublicAPI.Unshipped.txt]

    ~override Microsoft.Maui.Controls.Platform.Compatibility.ShellFlyoutRenderer.ViewWillTransitionToSize(CoreGraphics.CGSize toSize, UIKit.IUIViewControllerTransitionCoordinator coordinator) -> void

[src\Controls\src\Core\Compatibility\Handlers\Shell\iOS\ShellFlyoutRenderer.cs]

    public override void ViewWillTransitionToSize(CoreGraphics.CGSize toSize, UIKit.IUIViewControllerTransitionCoordinator coordinator)
    {
        base.ViewWillTransitionToSize(toSize, coordinator);

        coordinator.AnimateAlongsideTransition((IUIViewControllerTransitionCoordinatorContext obj) => 
        {
            // Before rotation
        }, 
        (IUIViewControllerTransitionCoordinatorContext obj) => 
        {
            // After rotation
            if (IsOpen)
            {
                TapoffView.Frame = View.Frame;
            }
        });
    }

Below is the execution result.

https://github.com/dotnet/maui/assets/125236133/d46b7d21-04f0-4b4a-9c51-f39f3db3d29b

Could you please check it? Thank you.

### Issues Fixed

<!-- Please make sure that there is a bug logged for the issue being fixed. The bug should describe the problem and how to reproduce it. -->

Fixes #15173

<!--
Are you targeting main? All PRs should target the main branch unless otherwise noted.
-->

cat0363 committed:
Added size update for Shell Popup Overlay after rotation

# Conflicts:
#	src/Controls/src/Core/PublicAPI/net-ios/PublicAPI.Unshipped.txt
#	src/Controls/src/Core/PublicAPI/net-maccatalyst/PublicAPI.Unshipped.txt

cat0363 committed:
Fix resizing on rotation

cat0363 commented:
The solution I posted last time is not beautiful because the overlay is temporarily drawn in the size 
before rotation when the terminal is rotated. To solve this problem, set the post-rotation size to 
TapoffView before rotation.

    public override void ViewWillTransitionToSize(CGSize toSize, IUIViewControllerTransitionCoordinator coordinator)
    {
        base.ViewWillTransitionToSize(toSize, coordinator);

        coordinator.AnimateAlongsideTransition((IUIViewControllerTransitionCoordinatorContext obj) =>
        {
            // Before rotation
            if (IsOpen)
            {
                TapoffView.Frame = new CGRect(View.Frame.X, View.Frame.Y, View.Frame.Width, View.Frame.Height);
            }
        }, null);
    }

Below is the execution result.

<table>
  <tr>
    <td>[Before Fix]</td>
    <td>[After Fix]</td>
  </tr>
  <tr>
    <td><video controls width="300" src="https://github.com/dotnet/maui/assets/125236133/bdc0999b-0937-4c3a-8e3e-ebd02ef906a2" /></td>
    <td><video controls width="300" src="https://github.com/dotnet/maui/assets/125236133/451e46d5-7cf3-401e-aae1-40d43422768d" /></td>
  </tr>
</table>

With this fix, overlays are no longer drawn at their pre-rotation size.

cat0363 commented:
@samhouts, A stale tag has been added, but has the issue been resolved?
Could you please tell me in which version the issue was resolved?

samhouts commented:
@cat0363 The stale label is just an easy way for us to query PRs that haven't been updated in a while. This isn't resolved to my knowledge because this PR hasn't been merged. Thanks for your patience!

cat0363 commented:
@samhouts , Thank you for your reply despite your busy schedule.
Your comment made me understand that the stale label serves as a marker.
I was worried that this label would cause it to be closed if no action was taken.
I will wait until the review of this PR starts.

PureWeen commented:
/azp run

PureWeen committed:
- fix publicapi

# Conflicts:
#	src/Controls/src/Core/PublicAPI/net-ios/PublicAPI.Unshipped.txt
#	src/Controls/src/Core/PublicAPI/net-maccatalyst/PublicAPI.Unshipped.txt

PureWeen commented:
/azp run

PureWeen commented:
/rebase

PureWeen committed:
- apply copilot suggested change

PureWeen committed:
- Add Tests

PureWeen committed:
- fix test

PureWeen committed:
- make backdrop orange
